From 4139fa09a00d253557b4f4c0b3f7dc6d8cedc277 Mon Sep 17 00:00:00 2001
From: Dinesh Kumar Mirche <dmirche@habana.ai>
Date: Tue, 13 May 2025 06:49:02 +0300
Subject: [PATCH] [SW-222126] gaudi3: sec: clear all aggregator register

As a part of security init in preboot, stale pcie sei
interrupts are cleared. This patch clears all three
interrupt aggregator registers for sei instead of only
one register.

Change-Id: Ibd9217d6ec5cc6567e465274215e25852d3bb200
Signed-off-by: Dinesh Kumar Mirche <dmirche@habana.ai>
---

diff --git a/arch/gaudi3/common/security.c b/arch/gaudi3/common/security.c
index 9020437..fbb6000 100644
--- a/arch/gaudi3/common/security.c
+++ b/arch/gaudi3/common/security.c
@@ -44,6 +44,7 @@
 #include <mstr_if_xresp_lbw_masks.h>
 #include <pcie_wrap_regs.h>
 #include <pcie_wrap_masks.h>
+#include <int_agg_shared_sei_int_msg_regs.h>
 #include <fw_info.h>
 #include <hl_sec_if.h>
 #include <telemetry_ops.h>
@@ -462,8 +463,17 @@
 	CFG_RREG32(mmD0_PCIE_WRAP_BASE + mmPCIE_WRAP_PCIE_SEI_INTR_STATUS);
 
 	/* clear the interrupt at aggregator */
-	CFG_WREG32(mmD0_CPU_INT_AGG_SHARED_SEI_INT_MSG_BASE,
-				CFG_RREG32(mmD0_CPU_INT_AGG_SHARED_SEI_INT_MSG_BASE));
+	CFG_WREG32((mmD0_CPU_INT_AGG_SHARED_SEI_INT_MSG_BASE + mmINT_AGG_SHARED_SEI_INT_MSG_STS_0),
+				CFG_RREG32(mmD0_CPU_INT_AGG_SHARED_SEI_INT_MSG_BASE +
+							mmINT_AGG_SHARED_SEI_INT_MSG_STS_0));
+
+	CFG_WREG32((mmD0_CPU_INT_AGG_SHARED_SEI_INT_MSG_BASE + mmINT_AGG_SHARED_SEI_INT_MSG_STS_1),
+				CFG_RREG32(mmD0_CPU_INT_AGG_SHARED_SEI_INT_MSG_BASE +
+							mmINT_AGG_SHARED_SEI_INT_MSG_STS_1));
+
+	CFG_WREG32((mmD0_CPU_INT_AGG_SHARED_SEI_INT_MSG_BASE + mmINT_AGG_SHARED_SEI_INT_MSG_STS_2),
+				CFG_RREG32(mmD0_CPU_INT_AGG_SHARED_SEI_INT_MSG_BASE +
+							mmINT_AGG_SHARED_SEI_INT_MSG_STS_2));
 }
 
 #define AXPROT_PERM_BITS	3
